### PR TITLE
Updated API docs

### LINKED ISSUE #1680: Review and update docs/api-reference
The [API Reference](https://www.pilosa.com/docs/latest/api-reference/) in docs needs to be updated. In particular, there are `index` and `field` options such as `keys` and `trackExistence` which need to be included and documented.

There are likely other http endpoints that we should document here as well.

### PR DESCRIPTION
**Pull request by yuce**
## Overview

Fixes #1680 

## Pull request checklist

- [ ] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [ ] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [ ] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [ ] I have resolved any merge conflicts.
- [ ] I have included tests that cover my changes.
- [ ] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

### CODE REVIEW

File: docs/api-reference.md [hunk 1]
```diff
@@ -17,9 +17,42 @@ Returns the schema of all indexes in JSON.
...
+                "trackExistence": true
+            }
+        }
+    ]
+}
 ```
 
+`GET /schema`
+
+Is equivalent to `GET /index` and returns the same response.
```
Comment (travisturner) [inline docs/api-reference.md]: It seems like we should apply the full description to the `/schema` docs, and put this "equivalent" description under the `/index` docs.

File: docs/api-reference.md [hunk 2]
```diff
@@ -39,8 +88,13 @@ curl -XGET localhost:10101/index/user
 
 Creates an index with the given name.
 
+The request payload is in JSON, and may contain the `options` field. The `options` field is a JSON object with the following options:
+
+* `keys` (bool): Enables using column keys instead of column IDs.
+* `trackExistence` (bool): Enables or disables existence tracking on the index. Required for [Not](../query-language/#not) queries. It is `true` by default. Note that disabling track existence improves query performance.
```
Comment (travisturner) [inline docs/api-reference.md]: `Note that disabling track existence improves query performance.` That's not necessarily a true statement. Disabling `trackExistence` would improve write performance, but not necessarily query performance. I think if we want to include anything like this, it should be more along the lines of "Note that writing to an index with `trackExistence` enabled incurs a small performance penalty."

File: docs/api-reference.md [hunk 3]
```diff
@@ -100,7 +177,12 @@ By default, all bits and attributes (*for `Row` queries only*) are returned. In
 
 Creates a field in the given index with the given name.
 
-The request payload is in JSON, and may contain the `options` field. The `options` field is a JSON object which must contain a `type` along with the corresponding configuration options. 
+The request payload is in JSON, and may contain the `options` field. The `options` field is a JSON object which must contain a `type` and optionally the `keys`:
```
Comment (travisturner) [inline docs/api-reference.md]: remove "and optionally the `keys`"
perhaps instead we can include "(optional)" after the `keys` description.

File: docs/api-reference.md [hunk 4]
```diff
@@ -171,6 +253,33 @@ curl -XGET localhost:10101/version
 {"version":"v0.6.0"}
 ```
 
+### Get status
+
+`GET /status`
+
+Returns the status of nodes in a Pilosa cluster.
```
Comment (travisturner) [inline docs/api-reference.md]: `/status` actually returns the status of the cluster. It includes the list of nodes in the cluster, but currently we don't provide a "node status".

**Comment by yuce** [comment]
@travisturner Is this ok to merge?

**Pull request merged**